book/scale_nodes.rst

```diff
@@ -0,0 +1,355 @@
+               entity_manager: default
+               document_manager: default
+           result_cache:
+               client: doctrine
+               entity_manager: [default]
+           query_cache:
+               client: doctrine
+               entity_manager: default
+
+5. **Sessions storage**. By default the parameters.yml is configured to store the session in files 
+   (*session_handler: *session.handler.native_file*). The simplest way to deal with the sessions in a web farm 
+   configuration is to store them in the DB - just change the value to "*session.handler.pdo*". 
+
+   You can also store them in Redis:
+
+**app/config/config.yml:**
+   
+.. code-block:: yaml
+
+   snc_redis:
+       clients:
+           ...
+           session:
+               type: phpredis
+               alias: session
+               dsn: redis://localhost/3
+       session:  # configure sessions
+           client: session
+           prefix: session
+           use_as_default: true
+
+6. There are two abstract services used in the Oro Platform that are responsible for the cache saving.
+   *oro.file_cache.abstract*: should be used to cache the data private for each node in a web farm
+   *oro.cache.abstract*: should be used to cache the data to be shared between the nodes.
+   
+   In our case the oro.cache.abstract* is applicable.
+
+**app/config/config.yml:**   
+
+.. code-block:: yaml
+   
+   services:
+       oro.cache.abstract:
+           class: Snc\RedisBundle\Doctrine\Cache\RedisCache
+           calls:
+               - [setRedis, [@snc_redis.default]]
+
+
+7. **Annotations** cache configuration:
+
+**app/config/config.yml:**
+
+.. code-block:: yaml
+
+   framework:
+       annotations:
+           cache: oro.cache.abstract
+
+8. **Doctrine** cache configuration:
+
+**app/config/config.yml:**
+
+.. code-block:: yaml
+
+   doctrine:
+       orm:
+           query_cache_driver: redis
+           result_cache_driver: redis
+           #metadata_cache_driver does NOT require any changes because it is already a child of oro.cache.abstract
+           #metadata_cache_driver:
+           #            type: service
+           #            id: doctrine.metadata.cache
+           #
+           #doctrine.metadata.cache: 
+           #        parent: oro.cache.abstract
+           #        calls:
+           #           - [ setNamespace, [ 'doctrine.metadata.cache' ] ]
+
+
+9. **JMS Serializer** cache configuration:
+
+**app/config/config.yml:**
+
+.. code-block:: yaml
+   
+   jms_serializer:
+       metadata:
+           cache: Metadata\Cache\DoctrineCacheAdapter
+
+10. **Security nonces**. Since each nonce should be used only once, they cannot be stored on every server, as we need
+    access to all the nonces when the next API request comes in.
+
+**app/config/security.yml:**
+
+.. code-block:: yaml
+
+   security:
+       firewalls:
+           wsse_secured:
+               wsse:
+                   nonce_cache_service_id: oro.cache.abstract
```

Comment:
create a concrete service based on `oro.cache.abstract` and use it instead of `oro.cache.abstract`. Example of a concrete service:
```
    oro_translation.cache:
          parent: oro.cache.abstract
          calls:
              - [setNamespace, [ 'oro_trans' ] ]
```